src/logics/graphImpl/visImpl.ts

```diff
@@ -12,7 +12,19 @@ export const layoutOptions = ['force-directed']
 let network: Network | null = null;
 const nodes = new DataSet<Node>({})
 const edges = new DataSet<Edge>({})
+let shiftKeyDown = false;
+
 const defaultOptions: Options = {
+  manipulation: {
+    addEdge: function (data: any, _callback: any) {
+      const edgeTo = data.to;
+      const edgeFrom = data.from;
+      store.dispatch(setIsNodeDialog(false));
+      store.dispatch(setEdgeFrom(edgeFrom));
+      store.dispatch(setEdgeTo(edgeTo));
+      store.dispatch(openDialog());
```

Comment:
This seems like a lot of unnecessary function calls. You can probably pass a lot of these as parameters to openDialog.
e.g. something similar to
```
openEdgeDialog(edgeFrom, edgeTo);
openNodeDialog(x, y);
````
or even
```
openDialog({type: 'edge', from: fromEdge, to: toEdge});
openDialog({type: 'node', x: pos.x, y: pos.y});
...
function openDialog(dialogState: NodeDialog| EdgeDialog) {
  return {...state, ...dialogState}
}
```